### ISSUE TITLE
Tool table offsets are not persistent over power cycle

### ISSUE THREAD

**Issue by karoria** [grblHAL/core#494]
Hi @terjeio 
I just checked with my new build that tool table offsets (of Z that is only written) are not saved persistently over power cycle(I get it by "$#" command). Is there any setting or compile time option I am missing?
Further, is there any command/variable to read particular tool's Z offset?
Regards,
Ravi

**Comment by karoria** [comment]
Hi @terjeio 
Can u reproduce this in your system? I use teensy 4.1 with custom board map.

**Comment by terjeio** [comment]
I cannot reproduce this, neither wih flash or EEPROM storage. What is your N_TOOLS config?

**Comment by karoria** [comment]
N_TOOLS is 6. I am using 5 axis configuration.

**Comment by terjeio** [comment]
> N_TOOLS is 6. I am using 5 axis configuration.

Works for me, only tested using flash storage. Settings and offsets are kept over a power cycle?

> Further, is there any command/variable to read particular tool's Z offset?

The only way is to apply it and calculate the change in position. Add an inbuilt  G65 macro to read it?

**Comment by karoria** [comment]
It looks like the tool table offsets given by G10L10P1Z10 command are not persistent over power cycle while G10L1P1Z10 are persistent over power cycle. Can you check it?
Which method do you recommend to save only Z offsets?

**Comment by terjeio** [comment]
L10 \(and L11\) does not store the changes, this could be incorrect behaviour though. The LinuxCNC documentation is, IMO, ambiguous in that the [L1](http://linuxcnc.org/docs/2.8/html/gcode/g-code.html#gcode:g10-l1) documentation excplicitly states that the tool table is written out but [L10 \(and L11\)](http://linuxcnc.org/docs/2.8/html/gcode/g-code.html#gcode:g10-l10) does not do so.
Somebody that runs LinuxCNC can check?

**Comment by MCULJ** [comment]
Does it support radius compensation?

**Comment by terjeio** [comment]
> Does it support radius compensation?

No. The radius is parsed out and stored - but [not yet used](https://github.com/grblHAL/core/blob/596576763e88f2bcd288b5f3d729d0b1cb1e85af/gcode.h#L608).
